### PR TITLE
Populate Highlight search on click (Issue #500)

### PR DESCRIPTION
**Pull request by gabrielzezze**
This tries to add the requested feature described in #500.

The `highlightQuery` data was already available and being used in the  `FlameGraphRenderer` so no major changes were needed.

Any questions, tips or issues I'll be happy to help!
(Sorry for the bad quality GIF the converter made it worse)

pyroscope-issue500

### CODE REVIEW

**Review by eh-am — CHANGES_REQUESTED** [review]
Beautiful! I didn't realize how useful this would be until seeing this PR!


I think from UX there's few things missing:
* some feedback that that's actually clickable! probably just a `cursor: pointer` on hover?
* some way to quickly clean that search image, this is something we already needed, but now that the whole interaction happens ONLY via mouse, it makes sense undoing that interaction via mouse as well.
* maybe highlighting in the table as well
image, not so sold on that, let me know what you think.

As always, you don't have to implement all of them. We can always break down into smaller tasks so anybody can come and help!


---
Tests failing:
From what I see tests are failing because we do a type check there. Which we DO NOT do when developing (for a couple reasons, a) it's faster and b) types getting in the way while prototyping is bit of a pain in the ass).
So it should be just a matter of adding the missing `highlightQuery` to `Toolbar` tests.

That being said, I think it's very important to add a test here. It doesn't have to be ultra fancy, you just click on a item in the table and `expect` that the `handleTableItemClick` handler was called with the correct parameter.

Let me know if you need any help here, as there's currently no tests for `ProfilerTable`, it will require some setup. Really, let me know if you need assistance here!


---
Other thoughts, you don't have to do anything about it:

As mentioned in this PR: https://github.com/pyroscope-io/pyroscope/issues/526#issue-1052526045, for highly nested/small sample it's difficult to find where that call is in the flamegraph:
image

**Comment by eh-am** [comment]
And RE: the cypress test that failed, you can check its logs here https://github.com/pyroscope-io/pyroscope/runs/4281141822?check_suite_focus=true and its diffs here https://github.com/pyroscope-io/pyroscope/actions/runs/1488134079
You can download the `cypress-snapshots` and look for the folder `__diff__`. It will show what it was expecting and what it got. For example: 

pyroscope server alloc_objects-table diff

Which indeed it seems the ellipsis are broken. This is `demo-dev` (basically what's in `main`):
image

And here's this PR:
image


So these tests are a bit of pain (to develop, to run, to update), but they help with cases like that!

File: webapp/javascript/components/ProfilerTable.jsx [hunk 1]
```diff
@@ -229,14 +240,19 @@ const TableBody = React.memo(
...
     const nameCell = (x, style) => (
       <td>
-        <span className="color-reference" style={style} />
-        <div
-          className="symbol-name"
-          title={x.name}
-          style={fitIntoTableCell(fitMode)}
+        <button
+          className="table-item-button"
+          onClick={handleTableItemClick ? () => handleTableItemClick(x) : null}
```
[the remark this answers is not in the document]
Comment (gabrielzezze) [reply webapp/javascript/components/ProfilerTable.jsx:245]: I do not think this is necessary, my bad... will resolve the conversation when removed.

File: webapp/javascript/components/FlameGraph/FlameGraphRenderer.jsx [hunk 2]
```diff
@@ -151,6 +126,10 @@ class FlameGraphRenderer extends React.Component {
     });
   };
 
+  onTableItemClick(tableItem) {
```
[the remark this answers is not in the document]
Comment (gabrielzezze) [reply webapp/javascript/components/FlameGraph/FlameGraphRenderer.jsx:129]: Sorry about that, I totally agree.

**Comment by gabrielzezze** [comment]
Hi @eh-am,

Thanks for the feedback, those ideas are great and I'll be happy to add them as well as the necessary tests.
About the failing `Toolbar` test, I'll be sure to fix it.

I apologize if I take a bit more time to resolve the problems related to the Cypress test as I'm still getting to know Pyroscope and don't quite get what's causing the issue as I didn't change anything related to the data.

**Comment by gabrielzezze** [comment]
Hi,

Just a heads up that I'm still working on this, had a tough week and could not make any progress but this week I have some extra free time to work on it.

Thanks!

**Comment by eh-am** [comment]
Hey @gabrielzezze, no worries! Last week was thanksgiving for the US folks so it was a slow week for us too :)

I think I forgot to tell you this, but I am available on our slack (https://pyroscope.io/slack), my name there is Eduardo Aleixo. Feel free to contact me!

**Comment by eh-am** [comment]
Closed in favour of https://github.com/pyroscope-io/pyroscope/pull/701

**Pull request closed without merging**